### PR TITLE
feat: limit green blockers at distance 2 for worlds

### PR DESCRIPTION
**Pull request by dalkia**
## What does this PR change?

Solves #3553 

The current Worlds implementation allows bigger scene areas than 2x2. But to get this started, a limit was implemented for any given scene at a distance of x Parcels, currently set at 2. 

Once the limit is applied to worlds, this will keep working.

Also, took the opportunity to due name refactoring (RealmsPlugin to RealmPlugin) as requested in the [Worlds ADR](https://fdc23f87.adr-cvq.pages.dev/adr/ADR-101)

## How to test the changes?

1. Enter `/changerealm menduz.dcl.eth` in the chat
2. You are now in a World; a reef. Green blockers should be 2 parcels away. No empty parcel should be loaded (meaning, no parcel with trees or rocks). All should be empty except for the reef
4. Go back to `/changerealm loki`. Walk to load any parcel. Green blockers should work normally
5. Enter Decentralnd using the link https://play.decentraland.zone/?renderer-branch=feat/limit-green-blockers-at-distance-2&realm=menduz.dcl.eth. Green blockers should be 2 parcels away. 

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### CODE REVIEW

**Review by menduz — CHANGES_REQUESTED** [review]
Can we enable/disable this behavior with an extra configuration or feature flag?

Remember that worlds will be used to host the onboarding experience and that is bigger than 2x2

**Review by menduz — COMMENTED** [review]
Can we enable/disable this behavior with an extra configuration or feature flag?

Remember that worlds will be used to host the onboarding experience and that is bigger than 2x2

**Review by kwspgoliasz — APPROVED** [review]
The feature appears to be implemented correctly.

File: unity-renderer/Assets/Scripts/MainScripts/DCL/WorldRuntime/BlockerController/WorldBlockersController.cs [hunk 1]
```diff
@@ -163,8 +167,27 @@ internal void SetupWorldBlockers(HashSet<Vector2Int> allLoadedParcelCoords)
                 }
             }
 
+            blockersToAdd = LookForLimits(allLoadedParcelCoords, blockers, 0);
+
+            // Remove extra blockers
+            foreach (var coords in blockersToRemove)
+            {
```
Comment (popuz) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/WorldRuntime/BlockerController/WorldBlockersController.cs:174]: you can remove the brackets

File: unity-renderer/Assets/Scripts/MainScripts/DCL/WorldRuntime/BlockerController/WorldBlockersController.cs [hunk 2]
```diff
@@ -163,8 +167,27 @@ internal void SetupWorldBlockers(HashSet<Vector2Int> allLoadedParcelCoords)
...
+                blockerInstanceHandler.HideBlocker(coords, false);
+            }
+
+            // Add missing blockers
+            foreach (var coords in blockersToAdd)
+            {
```
Comment (popuz) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/WorldRuntime/BlockerController/WorldBlockersController.cs:180]: can remove the brackets here as well

File: unity-renderer/Assets/Scripts/MainScripts/DCL/WorldRuntime/BlockerController/WorldBlockersController.cs [hunk 3]
```diff
@@ -173,27 +196,30 @@ internal void SetupWorldBlockers(HashSet<Vector2Int> allLoadedParcelCoords)
...
 
-            // Remove extra blockers
-            foreach (var coords in blockersToRemove)
+            if (currentLimitIterationEvaluation == worldBlockersLimit.Get())
+                return blockersCandidate;
+            else
             {
-                blockerInstanceHandler.HideBlocker(coords, false);
+                blockersCandidate.UnionWith(dontAddABlockerHere);
+                return LookForLimits(blockersCandidate, blockers, currentLimitIterationEvaluation + 1);
```
Comment (popuz) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/WorldRuntime/BlockerController/WorldBlockersController.cs:216]: I would probably just inline it inside and use ternary operator, but it is very optional (depends on your taste :) )
```
return currentLimitIterationEvaluation == worldBlockersLimit.Get()
                ? blockersCandidate
                : LookForLimits(blockersCandidate.UnionWith(dontAddABlockerHere), blockers, currentLimitIterationEvaluation + 1);

```

File: unity-renderer/Assets/DCLPlugins/RealmsPlugin/RealmBlockerModifier.cs [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+            bool shouldGreenBlockersBeActive = isCatalyst || ShouldGreenBlockersBeActive(realmConfiguration.Configurations.CityLoaderContentServer);
+            if (DataStore.i.featureFlags.flags.Get().IsFeatureEnabled(ENABLE_GREEN_BLOCKERS_WORLDS_FF))
+                DataStore.i.worldBlockers.worldBlockerLimit.Set(shouldGreenBlockersBeActive ? 0 : WORLD_BLOCKER_LIMIT);
+            else
+                DataStore.i.worldBlockers.worldBlockerEnabled.Set(shouldGreenBlockersBeActive);
+        }
+
+        public void Dispose() { }
+
+        private bool ShouldGreenBlockersBeActive(string cityLoaderContentServers)
```
Comment (popuz) [inline unity-renderer/Assets/DCLPlugins/RealmsPlugin/RealmBlockerModifier.cs:24]: we put `=>` on the same line in our code style (and only the body is on the next one)

File: unity-renderer/Assets/DCLPlugins/RealmsPlugin/RealmBlockerModifier.cs [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+using Decentraland.Bff;
+using JetBrains.Annotations;
+
+namespace DCLPlugins.RealmPlugin
+{
+    public class RealmBlockerModifier : IRealmModifier
+    {
+
+        private const int WORLD_BLOCKER_LIMIT = 2;
+        [CanBeNull] private const string ENABLE_GREEN_BLOCKERS_WORLDS_FF = "realms_blockers_in_worlds";
```
Comment (popuz) [inline unity-renderer/Assets/DCLPlugins/RealmsPlugin/RealmBlockerModifier.cs:11]: why do we need this `[CanBeNull]` attribute before the constant?

**Review by popuz — CHANGES_REQUESTED** [review]

File: unity-renderer/Assets/DCLPlugins/RealmsPlugin/RealmBlockerModifier.cs [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+            if (DataStore.i.featureFlags.flags.Get().IsFeatureEnabled(ENABLE_GREEN_BLOCKERS_WORLDS_FF))
+                worldBlockersLimit.Set(shouldGreenBlockersBeActive ? 0 : WORLD_BLOCKER_LIMIT);
+            else
+                worldBlockersEnabled.Set(shouldGreenBlockersBeActive);
+        }
+
+        public void Dispose() { }
+
+        private bool ShouldGreenBlockersBeActive(string cityLoaderContentServers) =>
```
Comment (popuz) [inline unity-renderer/Assets/DCLPlugins/RealmsPlugin/RealmBlockerModifier.cs:31]: I would rename to something more related to the check itself - maybe "HasContent"/"HasContenServers"...

Comment (popuz) [reply unity-renderer/Assets/DCLPlugins/RealmsPlugin/RealmBlockerModifier.cs:31]: You can also make it a local function

File: unity-renderer/Assets/DCLPlugins/RealmsPlugin/RealmBlockerModifier.cs [hunk 7]
```diff
@@ -0,0 +1,34 @@
+using DCL;
+using Decentraland.Bff;
+
+namespace DCLPlugins.RealmPlugin
+{
+    public class RealmBlockerModifier : IRealmModifier
```
Comment (popuz) [inline unity-renderer/Assets/DCLPlugins/RealmsPlugin/RealmBlockerModifier.cs:6]: What do you think about adding a short XML summary. Just to main responsibility of the class?

File: unity-renderer/Assets/DCLPlugins/RealmsPlugin/RealmMinimapModifier.cs [hunk 8]
```diff
@@ -0,0 +1,25 @@
+using DCL;
+using Decentraland.Bff;
+
+namespace DCLPlugins.RealmPlugin
+{
+    public class RealmMinimapModifier : IRealmModifier
```
Comment (popuz) [inline unity-renderer/Assets/DCLPlugins/RealmsPlugin/RealmMinimapModifier.cs:6]: Same here - "What do you think about adding a short XML summary (just to main responsibility of the class)?"

File: unity-renderer/Assets/DCLPlugins/RealmsPlugin/Tests/RealmBlockerModifierTests.cs [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+using System.Collections.Generic;
+
+namespace DCLPlugins.RealmPlugin
+{
+    public class RealmBlockerModifierTests
+    {
+        private RealmPlugin realmPlugin;
+        private const string CATALYST_REALM_NAME = "CatalystRealmName";
+        private const string WORLD_REALM_NAME = "WorldRealmName";
+        private const string ENABLE_GREEN_BLOCKERS_WORLDS_FF = "realms_blockers_in_worlds";
```
Comment (popuz) [inline unity-renderer/Assets/DCLPlugins/RealmsPlugin/Tests/RealmBlockerModifierTests.cs:14]: Can you please move constants up and then add blank line after them?

File: unity-renderer/Assets/DCLPlugins/RealmsPlugin/Tests/RealmPluginTests.cs [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+            DataStore.i.realm.playerRealmAboutConfiguration.Set(new AboutResponse.Types.AboutConfiguration()
+            {
+                RealmName = isWorld ? WORLD_REALM_NAME : CATALYST_REALM_NAME,
+                Minimap = new AboutResponse.Types.MinimapConfiguration()
+                {
+                    Enabled = !isWorld
+                },
+                ScenesUrn = { sceneUrn },
+            });
+        }
```
Comment (popuz) [inline unity-renderer/Assets/DCLPlugins/RealmsPlugin/Tests/RealmPluginTests.cs:56]: I saw this function in 3 Test classes. Would it be possible to move it to some static RealmPluginTestsUtils class (in the same folder)?

**Review by popuz — APPROVED** [review]
looks good, man 💪 🟢
just couple of cosmetical notes, as usual :)

File: unity-renderer/Assets/DCLPlugins/RealmsPlugin/RealmBlockerModifier.cs (hunk 6 above)
Comment (dalkia) [reply unity-renderer/Assets/DCLPlugins/RealmsPlugin/RealmBlockerModifier.cs:31]: Created as local function!

File: unity-renderer/Assets/DCLPlugins/RealmsPlugin/RealmBlockerModifier.cs (hunk 7 above)
Comment (dalkia) [reply unity-renderer/Assets/DCLPlugins/RealmsPlugin/RealmBlockerModifier.cs:6]: Done!

File: unity-renderer/Assets/DCLPlugins/RealmsPlugin/RealmMinimapModifier.cs (hunk 8 above)
Comment (dalkia) [reply unity-renderer/Assets/DCLPlugins/RealmsPlugin/RealmMinimapModifier.cs:6]: Done!

**Review by kwspgoliasz — APPROVED** [review]
The feature appears to be implemented correctly.

**Pull request merged**